crates/flipperzero/src/notification/mod.rs

```diff
@@ -0,0 +1,282 @@
+//! Furi dialogs.
+
+#[cfg(feature = "alloc")]
+use alloc::boxed::Box;
+
+use core::ffi::c_char;
+use core::ptr;
+
+use flipperzero_sys as sys;
+use flipperzero_sys::furi::{Status, UnsafeRecord};
+
+///Default notification messages.
+pub mod messages;
+///Default notification notes.
+pub mod notes;
+///Default notification sequences.
+pub mod sequences;
+
+const RECORD_NOTIFICATION: *const c_char = sys::c_string!("notification");
+
+/// A handle to the Notification app.
+pub struct NotificationApp {
+    data: UnsafeRecord<sys::NotificationApp>,
+}
+
+impl NotificationApp {
+    /// Obtains a handle to the Notifications app.
+    pub fn open() -> Self {
+        Self {
+            data: unsafe { UnsafeRecord::open(RECORD_NOTIFICATION) },
+        }
+    }
+
+    /// Runs a notification sequence.
+    #[cfg(feature = "alloc")]
+    pub fn notify<const N: usize>(&mut self, sequence: [&'static NotificationMessage; N]) {
+        unsafe {
+            sys::notification_message(
+                self.data.as_ptr(),
+                sequence
+                    .map(|msg| match msg {
+                        NotificationMessage::End => ptr::null(),
+                        _ => Box::leak(Box::new(msg.to_sys())) as *const _,
```

Comment:
This will constantly leak memory every time a notification is used.

If you make the internal representation of `NotificationMessage` match `sys::NotificationMessage`, then (because we have a `&'static NotificationMessage`) we can get a `&'static sys::NotificationMessage` as `*const sys::NotificationMessage` with the same static lifetime.